src/Hydrator.php

```diff
@@ -0,0 +1,245 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Hydrator;
+
+use Closure;
+use Psr\Container\ContainerInterface;
+use ReflectionAttribute;
+use ReflectionClass;
+use ReflectionProperty;
+use Yiisoft\Hydrator\Attribute\NoPopulate;
+use Yiisoft\Hydrator\Typecast\SimpleTypecast;
+use Yiisoft\Injector\Injector;
+
+use function array_key_exists;
+use function in_array;
+
+/**
+ * @psalm-import-type MapType from HydratorInterface
+ */
+final class Hydrator implements HydratorInterface
```

Comment:
What about caching?

Reply:
Caching what?

Reply:
Reflections for example

Reply:
Yes, its need, but later. Create issue, please.

Reply:
https://github.com/yiisoft/hydrator/issues/2